pkg/build/build.go

```diff
@@ -0,0 +1,334 @@
+package build
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/foomo/config-bob/pkg/providers"
+	"github.com/foomo/config-bob/pkg/templates"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	line = "-------------------------------------------------------------------------------"
+)
+
+type Configuration struct {
+	ValueFiles        []string
+	TemplatePaths     []string
+	TemplateFunctions template.FuncMap
+	SecretManager     providers.SecretProviderManager
+}
+
+func Build(l *zap.Logger, config Configuration) (Result, error) {
+	templateFunctions := templates.DefaultTemplateFunctions
+	if len(config.TemplateFunctions) != 0 {
+		templateFunctions = config.TemplateFunctions
+	}
+
+	templateFunctions["secret"] = config.SecretManager.GetSecret
+
+	l.Info(line)
+	fmt.Println("building")
+	fmt.Println("data files     :", strings.Join(config.ValueFiles, ", "))
+	fmt.Println("source folders :", strings.Join(config.TemplatePaths, ", "))
+	fmt.Println(line)
+	data, err := readData(config.ValueFiles)
+	if err != nil {
+		return Result{}, errors.New("could not read data from: " + strings.Join(config.ValueFiles, ", ") + " :: " + err.Error())
```

Comment:
Wrapf!

Reply:
`Wrapf` only makes sense when you need to inspect the error in the callee.